### PR TITLE
[OSDOCS-8077] OCP content port to ROSA and OSD:Web Console

### PR DESCRIPTION
**Pull request by mburke5678**
This PR ports the Web Console book from OCP to the OSD and ROSA books using single-sourcing.

[ROSA](https://66934--docspreview.netlify.app/openshift-rosa/latest/web_console/web-console-overview)
[OSD](https://66934--docspreview.netlify.app/openshift-dedicated/latest/web_console/web-console-overview)
[OCP](https://66934--docspreview.netlify.app/openshift-enterprise/latest/web_console/web-console-overview)

Issue: https://issues.redhat.com/browse/OSDOCS-8077

QE review:
- [x] QE has approved this change.

**Peer review: Very little new text. Most changes are the addition of `ifdef` tags.**

### CODE REVIEW

**Comment by mburke5678** [comment]
**NOTE TO SELF**
Replace link in `application-health.adoc`

**Comment by mburke5678** [comment]
@xueli181114 Please assign this PR to the most appropriate reviewers. Thank you!

**Comment by yanpzhan** [comment]
LGTM from QE side

File: modules/web-console-overview.adoc [hunk 1]
```diff
@@ -24,4 +27,18 @@ For existing clusters that you did not install, you can use `oc whoami --show-co
...
+
+ifdef::openshift-rosa,openshift-dedicated[]
+
+.Procedure
+
+. Log in to {cluster-manager-url} and click the name of your cluster.
+
+. On the cluster *Overview* tab, click *Open console*.
+
+. Log in with your credentials to access the cluster web console.
```
Comment (aireilly) [inline modules/web-console-overview.adoc:41]:
```suggestion
```

File: modules/web-console-overview.adoc [hunk 2]
```diff
@@ -24,4 +27,18 @@ For existing clusters that you did not install, you can use `oc whoami --show-co
...
+====
+endif::openshift-rosa,openshift-dedicated[]
+
+ifdef::openshift-rosa,openshift-dedicated[]
+
+.Procedure
+
+. Log in to {cluster-manager-url} and click the name of your cluster.
+
+. On the cluster *Overview* tab, click *Open console*.
```
Comment (aireilly) [inline modules/web-console-overview.adoc:39]:
```suggestion
. On the cluster *Overview* tab, click *Open console*, and log in with your credentials.
```

File: web_console/configuring-web-console.adoc [hunk 3]
```diff
@@ -13,5 +13,9 @@ or disable the console.
 
 * Deploy an {product-title} cluster.
 
+// cannot create resource "consoles"
```
Comment (aireilly) [inline web_console/configuring-web-console.adoc:16]: These comments seem quite terse and may be confusing for new writers. It is not clear what you are referring to. Who cannot create console resources?

File: web_console/configuring-web-console.adoc [hunk 4]
```diff
@@ -13,5 +13,9 @@ or disable the console.
...
+ifndef::openshift-rosa,openshift-dedicated[]
 include::modules/web-console-configuration.adoc[leveloffset=+1]
+// do not have sufficient permissions to read any cluster configuration
```
Comment (aireilly) [inline web_console/configuring-web-console.adoc:19]: Same comment as above re: clarity and meaning of these comments.

File: web_console/creating-quick-start-tutorials.adoc [hunk 5]
```diff
@@ -1,8 +1,17 @@
+ifndef::openshift-rosa,openshift-dedicated[]
 :_mod-docs-content-type: ASSEMBLY
 [id="creating-quick-start-tutorials"]
 = Creating quick start tutorials in the web console
 include::_attributes/common-attributes.adoc[]
 :context: creating-quick-start-tutorials
+endif::openshift-rosa,openshift-dedicated[]
```
Comment (aireilly) [inline web_console/creating-quick-start-tutorials.adoc:7]: Is the only difference here in the title? Suggest conditionalizing just the title if so.

File: web_console/creating-quick-start-tutorials.adoc [hunk 6]
```diff
@@ -14,6 +23,8 @@ include::modules/quick-start-user-workflow.adoc[leveloffset=+1]
 
 include::modules/quick-start-components.adoc[leveloffset=+1]
 
+// cannot create resource "consolequickstarts"
```
Comment (aireilly) [inline web_console/creating-quick-start-tutorials.adoc:26]: Same comment as above re: confusing comments

File: web_console/customizing-the-web-console.adoc [hunk 7]
```diff
@@ -26,26 +30,40 @@ If a resource has a message about resource limitations or quotas being reached,
 
 If using *List View* from the *View Shortcuts* menu, resources appear as a list. The *Alerts* column indicates if a message is available.
 
+// cannot create resource "secrets" in openshift-config
```
Comment (aireilly) [inline web_console/customizing-the-web-console.adoc:33]: Nit: I would prefer to see the repetitive comments consolidated and summarized for legibility.

File: web_console/creating-quick-start-tutorials.adoc (hunk 5 above)
Comment (aireilly) [reply web_console/creating-quick-start-tutorials.adoc:7]: Also, should the "Creating quick start tutorials in the web console" section be in the build output for openshift-enterprise? I don't see it. 

If at all possible, I would suggest not conditionalizing the topic title and preamble. It makes the content difficult to maintain.

**Review by aireilly — COMMENTED** [review]
Looks good, it is tricky to maintain this kind of heavy conditionalized content :)

Left a few nits and comments for your consideration.

File: modules/adding-yaml-examples-to-kube-resources.adoc [hunk 8]
```diff
@@ -10,7 +10,7 @@ You can dynamically add YAML examples to any Kubernetes resources at any time.
 
 .Prerequisites
 
-* You must have cluster administrator privileges.
+* You must have *cluster administrator* privileges.
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply modules/adding-yaml-examples-to-kube-resources.adoc:13]: Not sure why (or even if) I did that. Thanks!

File: web_console/creating-quick-start-tutorials.adoc (hunk 5 above)
Comment (mburke5678) [reply web_console/creating-quick-start-tutorials.adoc:7]: | Suggest conditionalizing just the title if so.

@aireilly I couldn't get it to work any other way than this.

Comment (mburke5678) [reply web_console/creating-quick-start-tutorials.adoc:7]: Here is the assembly in [OCP](https://66934--docspreview.netlify.app/openshift-enterprise/latest/web_console/creating-quick-start-tutorials).

File: _topic_maps/_topic_map_osd.yml [hunk 9]
```diff
@@ -189,6 +189,62 @@ Topics:
...
+- Name: Web console overview
+  File: web-console-overview
+- Name: Accessing the web console
+  File: web-console
+- Name: Viewing cluster information
+  File: using-dashboard-to-get-cluster-information
+- Name: Adding user preferences
+  File: adding-user-preferences
+  Distros: openshift-enterprise,openshift-origin
+# cannot patch resource "consoles", insufficient permissions to read any Cluster configuration
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply _topic_maps/_topic_map_osd.yml:205]: @aireilly We are leaving the irrelevant assemblies in in order to verify that the procedures within were all tested. the comments show why the assembly is hidden. I think folks will remove the commented-out material when the content port project is complete.

**Pull request merged**